rotkehlchen/chain/evm/decoding/structures.py

```diff
@@ -22,3 +23,30 @@ class ActionItem(NamedTuple):
     # Optional event data that pairs it with the event of the action item
     # Contains a tuple with the paired event and whether it's an out event (True) or in event
     paired_event_data: Optional[tuple[HistoryBaseEntry, bool]] = None
+
+
+@dataclass(init=True, repr=True, eq=True, order=False, unsafe_hash=False, frozen=False)
+class PostDecodingrules:
+    """
+    Define the different types of post decoding rules. There are some rules that only
+    need to be executed when certain decoders have been used, this is the case of balancer
+    post decoding rules for example but there are others that execute always and is by checking
+    the transaction address how we choose to run them or not (happens with uniswap).
+
+    This data class defines the field `common` for decoding rules that need to be executed
+    always and the field `with_counterparties` that will be later filtered to extract from it
+    the needed rules.
+    """
+    common: list[tuple[int, Callable]]
```

Comment:
Why common rules? I don't think they should ever exist. Post-decoding is inherently protocol specific. There should never be common post decoding rules.